From e2181627045d545acd5a138f8939209a12bf14a7 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Thu, 6 May 2021 17:09:43 -0700
Subject: [PATCH] gclue-config: read nmea-socket from config file

This option in the config file was being ignored, so this reads it.
Since the option is commented out 'by default', it's not a problem if it
can't be found in the config file (a debug message is printed to that
effect)
---
 src/gclue-config.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/gclue-config.c b/src/gclue-config.c
index 8e07910..7330762 100644
--- a/src/gclue-config.c
+++ b/src/gclue-config.c
@@ -84,6 +84,7 @@ gclue_config_finalize (GObject *object)
         g_clear_pointer (&priv->wifi_url, g_free);
         g_clear_pointer (&priv->wifi_submit_url, g_free);
         g_clear_pointer (&priv->wifi_submit_nick, g_free);
+        g_clear_pointer (&priv->nmea_socket, g_free);
 
         g_list_foreach (priv->app_configs, (GFunc) app_config_free, NULL);
 
@@ -299,8 +300,19 @@ load_modem_gps_config (GClueConfig *config)
 static void
 load_network_nmea_config (GClueConfig *config)
 {
+        GError *error = NULL;
         config->priv->enable_nmea_source =
                 load_enable_source_config (config, "network-nmea");
+        if (!config->priv->enable_nmea_source)
+                return;
+        config->priv->nmea_socket = g_key_file_get_string (config->priv->key_file,
+                                                           "network-nmea",
+                                                           "nmea-socket",
+                                                           &error);
+        if (error != NULL) {
+                g_debug ("`nmea-socket` configuration not set.");
+                g_clear_error (&error);
+        }
 }
 
 static void
-- 
GitLab

